src/org/zaproxy/zap/model/DefaultValueGenerator.java

```diff
@@ -0,0 +1,136 @@
+package org.zaproxy.zap.model;
+
+import java.net.URI;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Map;
+
+import net.htmlparser.jericho.FormControlType;
+
+public class DefaultValueGenerator implements ValueGenerator{
+	
+	private Date defaultDate;
+	private final String ATTR_TYPE = "type";
+	private final String DEFAULT_NUMBER_VALUE = "1";
+	private final String DEFAULT_TEXT_VALUE = org.parosproxy.paros.Constant.PROGRAM_NAME_SHORT;
+	private final String DEFAULT_PASS_VALUE = DEFAULT_TEXT_VALUE;
+	private final String DEFAULT_FILE_VALUE = "test_file.txt";
+	private final String DEFAULT_EMPTY_VALUE = "";
+	
+	/**
+	 * Gets the default {@code Date}, to be used for default values of date fields.
+	 *
+	 * @return the date, never {@code null}.
+	 * @see #setDefaultDate(Date)
+	 */
+	public Date getDefaultDate() {
+		if (defaultDate == null) {
+			return new Date();
+		}
+		return defaultDate;
+	}
+
+	/**
+	 * Sets the default {@code Date}, to be used for default values of date fields.
+	 * <p>
+	 * If none ({@code null}) is set it's used the current date when generating the values for the fields.

```

Comment:
`* If none ({@code null}) is set then the current date when generating the values for the fields is used.`